Sources/Moya/Task.swift

```diff
@@ -3,39 +3,74 @@ import Foundation
 /// Represents an HTTP task.
 public enum Task {
 
-    /// A request with no additional data.
-    case requestPlain
+    /// All different ways to set parameters in an HTTP request's body.
+    public enum BodyParams {
+        /// The given encodable will be set in the request's body without any additional encoding.
+        case raw(Data)
 
-    /// A requests body set with data.
-    case requestData(Data)
+        /// The given encodable will be json encoded in the request's body.
+        case json(Encodable, JSONEncoder = JSONEncoder())
 
-    /// A request body set with `Encodable` type
-    case requestJSONEncodable(Encodable)
+        /// The given encodable will be url encoded in the request's body.
+        case urlEncoded(Encodable, URLEncodedFormEncoder = URLEncodedFormEncoder())
+    }
 
-    /// A request body set with `Encodable` type and custom encoder
-    case requestCustomJSONEncodable(Encodable, encoder: JSONEncoder)
+    /// The given encodable will be encoded in the url's query.
+    public struct URLParams {
+        public var encodable: Encodable
```

Comment:
`let` instead of` var`?

Reply:
Those structs are just containers for those 2 properties. It could have been a tuple but I find it less clean with a tuple.
I think making it `var` allow a bit more flexibility when using it, even if I predict users should not have the need to mutate it.

Reply:
I think these shouldn't be mutable then.